src/ray/raylet/local_object_manager.cc

```diff
@@ -372,34 +360,40 @@ void LocalObjectManager::AsyncRestoreSpilledObject(
                  << object_url;
   if (is_external_storage_type_fs_ && spilled_objects_url_.count(object_id) == 0) {
     RAY_CHECK(!node_id.IsNil());
-    RAY_LOG(DEBUG)
-        << "Restoration request was ignored because the node didn't spill the object "
-        << object_id << " yet. It will be retried...";
-    // If the object wasn't spilled yet on this node, just return. The caller should retry
-    // in this case.
-    if (callback) {
-      callback(Status::OK());
-    }
+    callback(Status::PendingRequiredData("Object hasn't been spilled yet."));
     return;
   }
 
-  RAY_CHECK(objects_pending_restore_.emplace(object_id).second)
-      << "Object dedupe wasn't done properly. Please report if you see this issue.";
-  io_worker_pool_.PopRestoreWorker([this, object_id, object_url, callback](
+  auto it = objects_pending_restore_.find(object_id);
+  if (it != objects_pending_restore_.end()) {
+    // Restoration is already in progress, join that restoration by adding this request
+    // to the set of callbacks for that restoration.
+    it->second.push_back(callback);
+    return;
+  }
+  // Restoration is not yet in progress, so we proceed with starting a restoration.
+  it = objects_pending_restore_
```

Comment:
Why did you remove the check here?

Reply:
Because we short-circuit directly above this if an entry for `object_id` already exists in the map, so this should only be reached if there was was no entry for `object_id`. Besides, we're deduping in the pull manager now instead of here; multiple entries for `object_id` are supported due to us wanting to support multiple concurrent remote restoration requests.